### DISCUSSION TITLE
Passing different object types?

### DISCUSSION THREAD

**Discussion by paulzwarts** [skrashevich/double-take#595] [Q&A]
Good day all.  Love my Frigate/DT integration. I'm very glad to see the project still survives under care.

Is it possible to pass different object types from Frigate to DT? Person events seem to be a hardcoded expectation. You can subscribe to more MQTT topics but it seems they are ignored. My use case is rural and I want to train models for ID of my flock of chickens vs intruders such as unwanted fowl, or even predatorial threats like fox, raccoon, badger, etc. 

Under a default YOLO model, chickens are classified as birds, which as a starting point is perfect for dividing between friendlies and threats.. I'd like to use the Frigate/DT integration to speed up the queuing of annotation for your friendlies (your animals)  that you don't want a million clips and snapshots of. Being able to add attributes to the object type can be used in some useful telemetry research.

My desired behaviour would be to have DT queue up any object type configured in the objects config of frigate or DT. In my immediate case I would want specified objects to appear in DT for review and annotation. In the beginning, I can use bird, but later build a model correctly inferencing chicken from bird, fox from dog, etc. The main thing useful here is if event determines a threat is within certain zones, you immediately account for everything in the telemetry.

So right now DT announces at startup that its watching the MQTT/frigate messages for bird objects. But DT is ignoring and not putting them up for association. Did I do something incorrect, or is this only working for persons?

Addon question: Codeproject:AI seems to correctly identify the bounding box of the face over the bounding of the animal itself. So face detection should work?. How are the arbitary data relationships enforced in models? For instance a person has a face, a courier has a license plate and a package and delivery guy. Does Frigate and DT understand these entity relationships?

**Comment by lumloong** [comment]
You need Frigate train model.

**Comment by paulzwarts** [comment]
This doesn't make sense. The detector is codeproject.Ai .it already detects bird object type. I want these to appear in Dt for training and matching but it only look at person events. 



Paul Zwarts




---- On Thu, 25 Jul 2024 10:25:31 -0400 ***@***.*** wrote ----



You need Frigate train model.

**Reply by lumloong** [reply]
double-take is just face recognition, it's not the same principle. You can customize the training model directly on frigate that will do that. Coupled with HomeAssistant notifications.But the process will be complicated.

**Comment by paulzwarts** [comment]
So DT pulls a mqtt message and then displays the object as unmatched is because the dlface detector came up with unknown? Ok I understand, but I thank you for explaining it. 



Paul Zwarts




---- On Thu, 25 Jul 2024 11:10:34 -0400 ***@***.*** wrote ----



double-take is just face recognition, it's not the same principle. You can customize the training model directly on frigate that will do that. Coupled with HomeAssistant notifications.But the process will be complicated.
